### PR TITLE
chore: bump to nightly-2023-08-17

### PR DESCRIPTION
**Pull request by semorrison**
We need to adapt to simp failing if it makes no progress.

- [x] depends on https://github.com/leanprover/lean4/pull/2334
- [x] depends on https://github.com/leanprover/lean4/pull/2336
- [x] depends on #188

### CODE REVIEW

File: Std/Data/Int/DivMod.lean [hunk 1]
```diff
@@ -386,7 +386,6 @@ theorem emod_lt_of_pos (a : Int) {b : Int} (H : 0 < b) : a % b < b :=
   match a, b, eq_succ_of_zero_lt H with
   | ofNat m, _, ⟨n, rfl⟩ => ofNat_lt.2 (Nat.mod_lt _ (Nat.succ_pos _))
   | -[m+1], _, ⟨n, rfl⟩ => by
-    simp [emod, Int.subNatNat_eq_coe]
     exact Int.sub_lt_self _ (ofNat_lt.2 <| Nat.succ_pos _)
```
Comment (digama0) [inline Std/Data/Int/DivMod.lean:389]: unnecessary `by exact`

**Comment by digama0** [comment]
the PR has stuff from other PRs in it, looks like a botched merge

**Comment by eric-wieser** [comment]
See [this zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/difficulties.20with.20git.20rebase/near/385644322) for discussion about the accidental rebases.

**Comment by eric-wieser** [comment]
Using `git rebase` to remove four of the last 5 commits seems to do the job.

**Pull request merged**